### PR TITLE
optimized the reflective field fetch

### PR DESCRIPTION
**Pull request by sheepdreamofandroids**
cached the method once found
The assumption is that the underlying class won't change

### CODE REVIEW

File: rejoiner/src/main/java/com/google/api/graphql/rejoiner/ProtoToGql.java [hunk 1]
```diff
@@ -14,44 +14,28 @@
...
 import com.google.common.base.CharMatcher;
 import com.google.common.base.Converter;
-import com.google.common.base.Function;
 import com.google.common.collect.ImmutableList;
 import com.google.common.collect.ImmutableMap;
-import com.google.protobuf.Descriptors.Descriptor;
-import com.google.protobuf.Descriptors.EnumDescriptor;
-import com.google.protobuf.Descriptors.EnumValueDescriptor;
-import com.google.protobuf.Descriptors.FieldDescriptor;
+import com.google.protobuf.Descriptors.*;
```
Comment (siderakis) [inline rejoiner/src/main/java/com/google/api/graphql/rejoiner/ProtoToGql.java]: Please don’t use wildcard imports.

File: rejoiner/src/main/java/com/google/api/graphql/rejoiner/ProtoToGql.java [hunk 2]
```diff
@@ -264,4 +215,32 @@ static String getReferenceName(GenericDescriptor descriptor) {
   static GraphQLTypeReference getReference(GenericDescriptor descriptor) {
     return new GraphQLTypeReference(getReferenceName(descriptor));
   }
+
+  //  private java.util.function.Function<Object, Object> getter(Class clazz, String methodName) {
```
Comment (siderakis) [inline rejoiner/src/main/java/com/google/api/graphql/rejoiner/ProtoToGql.java]: This looks interesting, what is this?

File: rejoiner/pom.xml [hunk 3]
```diff
@@ -43,7 +43,7 @@
   <groupId>com.google.api.graphql</groupId>
   <artifactId>rejoiner</artifactId>
   <name>Rejoiner</name>
-  <version>0.2.1</version>
+  <version>0.3.1</version>
```
Comment (siderakis) [inline rejoiner/pom.xml]: I’m not sure why these changes are here, maybe syncing would remove them?

File: rejoiner/src/main/java/com/google/api/graphql/rejoiner/ProtoToGql.java (hunk 2 above)
Comment (sheepdreamofandroids) [reply rejoiner/src/main/java/com/google/api/graphql/rejoiner/ProtoToGql.java]: I should probably remove this.
I came along this article: https://dzone.com/articles/java-reflection-but-faster
LambdaMetafactory would be ever so slightly faster than reflection. I'm just not sure whether the slight performance improvement would be worth the extra complexity.
It's not working yet and I just didn't find the time to finish it.
